src/DotVVM.Framework/Hosting/LocalizablePresenter.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Globalization;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace DotVVM.Framework.Hosting
+{
+    public class LocalizablePresenter : IDotvvmPresenter
+    {
+        private readonly Func<IDotvvmRequestContext, Task> nextPresenter;
+        private readonly Func<IDotvvmRequestContext, CultureInfo> getCulture;
+
+        public LocalizablePresenter(
+            Func<IDotvvmRequestContext, CultureInfo> getCulture,
+            Func<IDotvvmRequestContext, Task> nextPresenter
+        )
+        {
+            this.getCulture = getCulture;
+            this.nextPresenter = nextPresenter;
+        }
+
+        public Task ProcessRequest(IDotvvmRequestContext context)
+        {
+            var culture = this.getCulture(context);
+            if (culture != null)
+                CultureInfo.CurrentCulture = CultureInfo.CurrentUICulture = culture;
+            return this.nextPresenter(context);
+        }
+
+        public static Func<LocalizablePresenter> BasedOnParameter(string name)
+        {
+            var presenter = new LocalizablePresenter(
+                context => context.Parameters.TryGetValue(name, out var value) && !string.IsNullOrEmpty(value as string) ? new CultureInfo((string)value) : null,
```

Comment:
if `value` is not a valid culture name the statement `new CultureInfo((string)value)` will throw `CultureNotFoundException`. We should put the statement into a `try` block.

Reply:
Hmm, I don't think so, silently eating the error may be pretty nasty. I think that displaying Error 500 when you access non-existent culture is not that bad option. Unfortunately, in the presenter it's too late for a "not found" response, you can only do that in a middleware, but you could catch the CultureNotFound exception and rewrite it to 404 response elsewhere.

Reply:
Yes, silently eating the error is pretty nasty, but IMHO 500 Internal Server Error is worse. Error 500 should indicate that something is bad with your code. We can add `culture` constraint but that doesn't solve the problem. Returning 404 would be nice, but IMHO we shouldn't require for user to implement this behavior.

Reply:
Maybe I could do a redirect to a default culture. What would you think about that?

Reply:
That sounds good.